src/Plugin.php

```diff
@@ -137,12 +137,23 @@ public static function purgePost($postId) {
   /**
    * Registers the updated media file and its derivative image sizes for purging.
    *
+   * Since the file key is added to image attachments only, ensure to register
+   * remaining mime types and possible preview images for purging.
+   *
    * @param array $attachment
    *   The data of the attachment to purge.
    * @param int $postId
    *   The ID of the post the media file is attached to.
+   *
+   * @see wp_generate_attachment_metadata()
    */
   public static function purgeAttachmentMeta($attachment, $postId) {
+    if (empty($attachment['file'])) {
+      $attachment = (array) wp_get_attachment_metadata($postId);
```

Comment:
I added the typecasting as the returned meta data doesn't seem to always be an array. E.g. I tested it with a zip file and it returned a string "2".

Reply:
Please put this info about the special case into a code comment - it's not obvious otherwise as it requires to repeat manual testing.